sections/design/why-spa.tex

```diff
@@ -0,0 +1,48 @@
+The frameworks used also makes fast development possible because of the powerful tools that are provided. 
+It is also easy to create a mobile app based on the finished code of an SPA \cite{SPAvsMPAMerehead}.
+SPAs also have greate caching possibilities and the local data can be used offline for when the user has problems with their connection. 
+The pages can then still be shown to the user even with connectivity issues and the local data will be synchronized once the connection is up again \cite{SPAvsMPARuby}.
+\\\\
+Disadvantages of an SPA is the poor optimization for search engines. Most of the pages will not be available to scan because of the way an SPA works. 
+Some users will also have JavaScript disabled in their browsers, and since the SPA is reliant on JavaScript to load new page content all the time, the pages will not be able to load correctly.
+Generally SPAs have less security but with use of the modern frameworks mentioned the security of the page is significantly improved.
+SPAs also can not take advantes of browser history since the back button will can not go to the previous page since there only is one. 
+There are solutions that can be used for this issue such as using a history API \cite{SPAvsMPAMerehead}.
+
+\subsection{MPAs}
+MPAs are different to SPAs in that they have a more standard architecture. When a new page is loaded it will send a request to the server and all the data will be updated even if it is just a small amount. A MPA will therefore spent a bit more time loading a new page because it has to update everything.
+Actions can be taken to increase the loading speed. 
+Libraries such as jQuery can be used.
+Another popular thing to do to reduce load times is use filters for searching through a list of items as filters does not require a reload of the page.
+MPAs are generally used for web applications with a number of pages that have static information such as text, images and links to other pages. The content of an MPA will be split into many pages with many links to other pages to allow for greater performance.
+\\\\
+Advantages of MPAs are their ability to optimize their pages for search engines which will make them more accessible to others. 
+They are also easy to develop on since the larger technologies, such as the frameworks mentioned for SPAs, are typically not needed \cite{SPAvsMPAMerehead}.
+MPAs have unlimited scalability since new content can be placed on new pages with no limitations where as SPAs are more limited to the amount of content that can be placed on a single page without having it affect the performance \cite{SPAvsMPARuby}.
+\\\\
+Disadvantages of MPAs are the slower performance that comes with having to load all the content every time a page is loaded.
+It will generally also take longer time to develop an MPA since its difficult to separate the frontend and backend as they are closely knit together.
```

Comment:
```suggestion
It will generally also take more time to develop an MPA since it is difficult to separate the frontend and backend as they are closely knit together.
```